### PR TITLE
Inlcude docstring of `make_napari_viewer` in `testing.md`

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues
Towards https://github.com/napari/docs/issues/150
Depends on https://github.com/napari/napari/pull/6774

# Description
Use `autofunction` to include docstring of `make_napari_viewer`

### CODE REVIEW

**Comment by lucyleeow** [comment]
I think this is ready for review now.
I tried to not include the full module path to `make_napari_viewer` but I couldn't figure out how to do this in just this instance. It's probably fine that the full path is rendered:

image

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]: Could we make this a proper admonition? It kinda looks like a code block.
I think this is actually pretty important, especially when learning, so a proper admonition would make it stand out more.

Comment (psobolewskiPhD) [reply docs/developers/contributing/testing.md]: and in the admonition i'd put the python module name pytest-qt in backticks.

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]:
```suggestion
`make_napari_viewer` as a test function parameter, as shown in the **Examples** section below:
```

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]: @lucyleeow Given your comment about showing the full path, is it worth noting here that one does not need to import it? It's implied here, but maybe not clear to a new person? (fixtures are pretty magical as noted above)

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
I have a few small nitpicks, but otherwise I like it.

**Comment by Czaki** [comment]
I think that we should more promote `make_napari_viewer_proxy` as it will provide wrapped viewer, same as will be passed to plugins widgets.

File: docs/developers/contributing/testing.md
Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: Will do and I've added that pytest-qt is a napari testing dependency.

File: docs/developers/contributing/testing.md
Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: Good point! Will do!

**Comment by lucyleeow** [comment]
Thanks @psobolewskiPhD , changes made!

@Czaki so this testing section is aimed at napari developers, there is a separate section on testing for plugin developers. Are you recommending `make_napari_viewer_proxy` to be used for plugin testing or napari testing?

I will make another PR to add cross references to other testing sections.

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]:
```suggestion
```{note}
```

A generic requires a title I think?

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md:209]:
```suggestion
```
```
Need to close the admonition with 3 backticks, not sure the suggestion will work.

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [reply docs/developers/contributing/testing.md]: Currently it looks like:
image

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]:
```suggestion
````{note}
```
Sorry! I missed the nested python blocks! need 4 backticks

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md:215]:
```suggestion
````
```
4 backticks to close

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]: don't think the indentation is needed if the admonition has 4 backticks.

File: docs/developers/contributing/testing.md
Comment (psobolewskiPhD) [reply docs/developers/contributing/testing.md]: image

File: docs/developers/contributing/testing.md
Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: Interesting, I'll give it a go. (note I haven't set up building the docs locally :grimacing:, I noped out at the 3 repos part)

Comment (lucyleeow) [reply docs/developers/contributing/testing.md]: I don't mind the indentation though, visually nicer? Can change though

Comment (psobolewskiPhD) [reply docs/developers/contributing/testing.md]: Hmm, looks like the indents need to go?
image
Happy to be corrected, but I have no other explanation anyways.

**Review by psobolewskiPhD — APPROVED** [review]
thanks for your patience!

**Comment by lucyleeow** [comment]
No thank you for your patience!

I just added a empty line at the end of the note because I thought the code block looked weird otherwise. Good to go if you're happy.

image

**Review by DragaDoncila — APPROVED** [review]
This looks good to me! Since this is for internal tests I don't think we should change this to use the `viewer` proxy.

**Comment by psobolewskiPhD** [comment]
I think the spacing/padding/margin -- if I am understanding correctly -- is fixed in the theme:
https://github.com/napari/napari-sphinx-theme/issues/100
But probably not released.

**Comment by lucyleeow** [comment]
> I think the spacing/padding/margin -- if I am understanding correctly -- is fixed in the theme:

@psobolewskiPhD in that case I am happy to remove the empty line? I did notice there are other places with similar situation, so we may as well be consistent with what is happening already.

**Comment by psobolewskiPhD** [comment]
❤️
Let's wait for the release and then re-assess.

**Pull request merged**
